File: modules/ia_over_the_counter/android/src/main/kotlin/de/ihreapotheken/appsdk/flutter/over_the_counter/IaSdkFlutterOverTheCounter.kt
package de.ihreapotheken.appsdk.flutter.over_the_counter

import de.ihreapotheken.sdk.integrations.api.IaSdk
import de.ihreapotheken.sdk.otc.OtcModule
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** IaSdkFlutterOverTheCounter */
class IaSdkFlutterOverTheCounter :
    FlutterPlugin,
    MethodCallHandler {

    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(
            flutterPluginBinding.binaryMessenger,
            "de.ihreapotheken/sdk/overTheCounter",
        )
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(
        call: MethodCall,
        result: Result
    ) {
        if (call.method == "register") {
            IaSdk.register(OtcModule)
            result.success(null)
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: modules/ia_over_the_counter/ios/ia_over_the_counter/Sources/ia_over_the_counter/IaSdkFlutterOverTheCounter.swift
import Flutter
import UIKit
import IACore
import IAOverTheCounter

@MainActor
public class IaSdkFlutterOverTheCounter: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
        name: "de.ihreapotheken/sdk/overTheCounter",
        binaryMessenger: registrar.messenger(),
    )
    let instance = IaSdkFlutterOverTheCounter()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "register":
        IASDK.register([IASDKModule.overTheCounter])
        result(nil)
    default:
        result(FlutterMethodNotImplemented)
    }
  }
}
